macros/schema_tests/distributional/expect_row_values_to_have_data_for_every_n_datepart.sql

```diff
@@ -42,12 +43,34 @@
 with base_dates as (
 
     {{ dbt_date.get_base_dates(start_date=start_date, end_date=end_date, datepart=date_part) }}
+    {% if interval %}
+    where mod(
+            cast({{ dbt_utils.datediff("'" ~ start_date ~ "'", 'date_' ~ date_part, date_part) }} as {{ dbt_utils.type_int() }}),
+            cast({{interval}} as {{ dbt_utils.type_int() }})
+        ) = 0
+    {% endif %}
 
 ),
 model_data as (
 
     select
+    {% if not interval %}
+
         cast({{ dbt_utils.date_trunc(date_part, date_col) }} as {{ dbt_expectations.type_datetime() }}) as date_{{ date_part }},
+
+    {% else %}
+
```

Comment:
Would be great to add comments, along with an example showing sample data, of what this section does.

Reply:
I added some comments but had some difficulty keeping them brief. Please edit as you see fit, and feel free to reach out with any questions